File: 02_TDD/01_FizzBuzz/java/02_solution/src/main/java/FizzBuzz.java
public class FizzBuzz {
    public static final String FIZZ = "Fizz";    
    public static final String BUZZ = "Buzz";
    public static final String FIZZBUZZ = FIZZ + BUZZ;
    
    public String get(Integer value) {
        if (isFizz(value) && isBuzz(value))
            return FIZZBUZZ;
        if (isFizz(value))
            return FIZZ;
        if (isBuzz(value))
            return BUZZ;
        return value.toString();
    }

    private boolean isFizz(Integer value) {
        return value % 3 == 0;
    }

    private boolean isBuzz(Integer value) {
        return value % 5 == 0;
    }

}

File: 03_FizzBuzz/ts/02_solution/src/fizzbuzz.ts
const FIZZ = "Fizz";
const BUZZ = "Buzz";
const FIZZBUZZ = FIZZ + BUZZ;

export function fizzbuzz(number: number): string {
  if (isFizz(number) && isBuzz(number)) return FIZZBUZZ;
  if (isFizz(number)) return FIZZ;
  if (isBuzz(number)) return BUZZ;
  return number.toString();
}

function isFizz(number: number) {
  return isMultipleOf(3, number);
}

function isBuzz(number: number) {
  return isMultipleOf(5, number);
}

function isMultipleOf(multiple: number, number: number) {
  return number % multiple === 0;
}
